### PR TITLE
Skipping e2e tests when the change only docs(*.md) #371

### LINKED ISSUE #371: Speed up CI by skipping e2e tests when the change obviously not relevant to that
**What would you like to be added**:

skipping e2e tests when the change obviously not relevant to that, i.e. document changes.

**Why is this needed**:

Speed up CI.

### PR DESCRIPTION
**Pull request by u5surf**
**What type of PR is this?**
/kind test

**What this PR does / why we need it**:
speed up ci
**Which issue(s) 
Fixes #371 

**Special notes for your reviewer**:
 I’d like to test on travis-ci, that’s why I made this PR. Previously, I could execute this ci how I just only push my branch.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -32,7 +32,13 @@ jobs:
       name: "integration test edge"     # names the fourth Tests stage job
     - script: make edge_cross_build
       name: "cross build edge"     # names the fifth Tests stage job
-    - script: make e2e_test
+    - script:
+      | if ! git diff --name-only $TRAVIS_COMMIT_RANGE | grep -qvE '(.md)'
```
Comment (m1093782566) [inline .travis.yml]: What will happen if a commit includes two changes: one for `foo.go` and the other one for `bar.md`?

**Comment by m1093782566** [comment]
@pavan187 Seems CI got stuck?

**Comment by pavan187** [comment]
> @pavan187 Seems CI got stuck?

Seems rebase is not done.

File: .travis.yml (hunk 1 above)
Comment (u5surf) [reply .travis.yml]: grep is quiet, invert-match, and extend-regexp mode. 
I think invert-match means if match except (.md) then it returns true.
And list of filenames reads OR conditions that even only one NOT .md file returns true. 
e.g.
grep -qvE foo.md is false
grep -qvE foo1.md is false
grep -qvE bar.go is true
OR conditions of filelist is true
So, invert-if statement pass through the else conditions that executes e2e test.
c.f. https://reflectoring.io/skip-ci-build/

Comment (m1093782566) [reply .travis.yml]: Okay, thanks for clarification. 

Per the blog, why not

 `if ! git diff --name-only $TRAVIS_COMMIT_RANGE | grep -qvE '(.md$)'`?

Comment (u5surf) [reply .travis.yml]: sorry, typo found.`$`

**Pull request merged**